pkg/sql/logictest/testdata/logic_test/row_level_ttl

```diff
@@ -1199,3 +1217,75 @@ statement ok
 DROP TABLE "Table-Name"
 
 subtest end
+
+subtest set_both_reset_ttl_expire_after
+
+statement ok
+CREATE TABLE tbl_set_both_reset_ttl_expire_after (
+  id INT PRIMARY KEY,
+  expire_at TIMESTAMPTZ,
+  FAMILY (id, expire_at)
+) WITH (
+  ttl_expire_after = '10m',
+  ttl_expiration_expression = 'expire_at'
+)
+
+statement ok
+ALTER TABLE tbl_set_both_reset_ttl_expire_after RESET (ttl_expire_after)
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE tbl_set_both_reset_ttl_expire_after]
+----
+CREATE TABLE public.tbl_set_both_reset_ttl_expire_after (
+  id INT8 NOT NULL,
+  expire_at TIMESTAMPTZ NULL,
+  CONSTRAINT tbl_set_both_reset_ttl_expire_after_pkey PRIMARY KEY (id ASC),
+  FAMILY fam_0_id_expire_at_crdb_internal_expiration (id, expire_at)
+)
+
+let $label_suffix
+SELECT relname || ' [' || oid || ']' FROM pg_class WHERE relname = 'tbl_set_both_reset_ttl_expire_after'
+
+query TTT
+SELECT schedule_status, recurrence, owner FROM [SHOW SCHEDULES]
+WHERE label = 'row-level-ttl: $label_suffix'
+----
+
+subtest end
+
+subtest set_both_reset_ttl_expiration_expression
+
+statement ok
+CREATE TABLE tbl_set_both_reset_ttl_expiration_expression (
+  id INT PRIMARY KEY,
+  expire_at TIMESTAMPTZ,
+  FAMILY (id, expire_at)
+) WITH (
+  ttl_expire_after = '10m',
+  ttl_expiration_expression = 'expire_at'
+)
+
+statement ok
+ALTER TABLE tbl_set_both_reset_ttl_expiration_expression RESET (ttl_expiration_expression)
+
+query T
+SELECT create_statement FROM [SHOW CREATE TABLE tbl_set_both_reset_ttl_expiration_expression]
+----
+CREATE TABLE public.tbl_set_both_reset_ttl_expiration_expression (
+  id INT8 NOT NULL,
+  expire_at TIMESTAMPTZ NULL,
+  crdb_internal_expiration TIMESTAMPTZ NOT VISIBLE NOT NULL DEFAULT current_timestamp():::TIMESTAMPTZ + '00:10:00':::INTERVAL ON UPDATE current_timestamp():::TIMESTAMPTZ + '00:10:00':::INTERVAL,
+  CONSTRAINT tbl_set_both_reset_ttl_expiration_expression_pkey PRIMARY KEY (id ASC),
+  FAMILY fam_0_id_expire_at_crdb_internal_expiration (id, expire_at, crdb_internal_expiration)
+) WITH (ttl = 'on', ttl_expire_after = '00:10:00':::INTERVAL, ttl_job_cron = '@hourly')
+
+let $label_suffix
+SELECT relname || ' [' || oid || ']' FROM pg_class WHERE relname = 'tbl_set_both_reset_ttl_expiration_expression'
+
+query TTT
+SELECT schedule_status, recurrence, owner FROM [SHOW SCHEDULES]
+WHERE label = 'row-level-ttl: $label_suffix'
+----
+ACTIVE  @hourly  root
+
+subtest end
```

Comment:
Overall, I didn't understand why we'd like to put this logictest here in this commit. Shouldn't we add such a logictest in the same commit that we fix the bug so that we can assert that resetting (ttl_expire_after) or resetting (ttl_expiration_expression) do not reset the TTL configuration, but instead only resetting (TTL) will?

Reply:
This subtest was part of another subtest (at least the part that tested the create statement), but it was asserting the wrong thing.

I think it is easier to understand the subtests when they are more focused so I added a commit that splits some subtests up.

Reply:
I agree with the point of splitting a previously large subtest into smaller ones. What I was saying is the last *newly added* subtest, "subtest set_both_reset_ttl_expire_after" and "set_both_reset_ttl_expiration_expression", shouldn't be in this commit. They should be in the commit where you actually make the fix.

Moreover, subtest "set_both_reset_ttl_expire_after" isn't the expected behavior right? As you said, we'd expect resetting "expire_after" to *not* remove the whole TTL configuration, but the test asserted otherwise (of course, we haven't made any actual code fixes yet so we get the current behavior).

Reply:
Talked about this offline, these were part of subtests above. The subtest that `tbl_set_both_reset_ttl_expire_after` was part of was asserting the wrong thing.